torch/nn/modules/conv.py

```diff
@@ -35,16 +36,31 @@ def __init__(self, in_channels, out_channels, kernel_size, stride,
             self.bias = Parameter(torch.Tensor(out_channels))
         else:
             self.register_parameter('bias', None)
+        self.initializer = initializer
         self.reset_parameters()
 
     def reset_parameters(self):
         n = self.in_channels
         for k in self.kernel_size:
             n *= k
         stdv = 1. / math.sqrt(n)
-        self.weight.data.uniform_(-stdv, stdv)
-        if self.bias is not None:
-            self.bias.data.uniform_(-stdv, stdv)
+
+        if isinstance(self.initializer, function):
```

Comment:
This looks a bit verbose. You can probably simplify it by using the right initializers when no default argument was passed?